theano/tensor/basic.py

```diff
@@ -1266,28 +1266,19 @@ class MaxAndArgmax(Op):
 
     def make_node(self, x, axis=None):
         x = _as_tensor_variable(x)
-
-        if isinstance(axis, (tuple, list)):
+        if isinstance(axis, (tuple, list, numpy.ndarray)):
+            # List of axes: make them non-negative, and sort them
             axis = [int(a) for a in axis]
-            if len(axis) != 1:
-                axis = list(axis)
-                for idx in xrange(len(axis)):
-                    if axis[idx] < 0:
-                        axis[idx] += x.type.ndim
-                axis.sort()
-                if axis == list(range(-x.type.ndim, 0, 1)):
-                    axis = list(range(x.type.ndim))
-                assert axis == list(range(x.type.ndim)), (
-                    "MaxAndArgmax does not support multiple"
-                    " axes. the max fct supports it. Got %s" % axis)
+            
+            #if axis == list(range(-x.type.ndim, 0, 1)):
+                #axis = list(range(x.type.ndim))
+            #assert axis == list(range(x.type.ndim)), (
+                #"MaxAndArgmax does not support multiple"
+                #" axes. the max fct supports it. Got %s" % axis)
+            if axis == list(range(x.type.ndim)):
```

Comment:
if axis is an ndarray, it will crash:

import numpy
>>> axis=numpy.arange(4)
>>> if axis == list(range(4)): print 3
... 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()

You must make a special case for ndarray.